### PR TITLE
[GPU] Disable predict_preallocation_shape for blocked format

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Disable predict_preallocation_shape for blocked format*
 - *Fix to use layout instead of shape and data_type*

### Tickets:
 - *132334*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp [hunk 1]
```diff
@@ -496,7 +496,10 @@ event::ptr primitive_inst::realloc_if_needed() {
 
                 _outputs[0] = variable.get_memory();
                 // To record shape predictor
-                auto prealloc_info = sp.predict_preallocation_shape(id(), _impl_params->output_layouts[0].get_shape(), dt_size, true);
+                auto prealloc_info = sp.predict_preallocation_shape(id(), cldnn::layout(_impl_params->output_layouts[0].get_shape(),
+                                                                                        actual_layout.data_type,
+                                                                                        _impl_params->output_layouts[0].format),
+                                                                    true);
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/primitive_inst.cpp:502]: Can't we just use `_impl_params->output_layouts[0]` here?
```suggestion
                auto prealloc_info = sp.predict_preallocation_shape(id(), _impl_params->output_layouts[0], true);
```

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp [hunk 2]
```diff
@@ -586,7 +589,10 @@ event::ptr primitive_inst::realloc_if_needed() {
         // If debug config is set, repsect the config most
         tmp_prealloc_count = -1;
     }
-    prealloc_info = sp.predict_preallocation_shape(id(), current_shape, dt_size, can_reuse_buffer, tmp_prealloc_count);
+    prealloc_info = sp.predict_preallocation_shape(id(), cldnn::layout(current_shape,
+                                                                       actual_layout.data_type,
+                                                                       updated_layout.format),
+                                                   can_reuse_buffer, tmp_prealloc_count);
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/primitive_inst.cpp:595]:
```suggestion
    prealloc_info = sp.predict_preallocation_shape(id(), updated_layout, can_reuse_buffer, tmp_prealloc_count);
```

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp (hunk 1 above)
Comment (steve-y) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp]: Updated as you commented, thanks.

**Comment by isanghao** [comment]
no issue from dgpu daily test

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**